src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java

```diff
@@ -0,0 +1,163 @@
+package com.github.princesslana.eriscasper.repository.event;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.event.ChannelCreateEvent;
+import com.github.princesslana.eriscasper.data.event.ChannelDeleteEvent;
+import com.github.princesslana.eriscasper.data.event.Event;
+import com.github.princesslana.eriscasper.data.event.GuildCreateEvent;
+import com.github.princesslana.eriscasper.data.event.GuildDeleteEvent;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import com.github.princesslana.eriscasper.data.resource.ImmutableChannel;
+import com.github.princesslana.eriscasper.repository.GuildRepository;
+import com.github.princesslana.eriscasper.rx.Maybes;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.observables.ConnectableObservable;
+import io.reactivex.subjects.PublishSubject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class GuildsFromEvents implements GuildRepository {
+
+  private final ConnectableObservable<ImmutableMap<Snowflake, Guild>> guildCache;
+  private final ConnectableObservable<ImmutableMap<Snowflake, Channel>> channelCache;
+
+  public GuildsFromEvents(Observable<Event> eventObservable) {
```

Comment:
This constructor feels large and complex. What could we do to clarify what it's doing?

Reply:
I mean the whole constructor, not the parameter name.

e.g., Is there stuff we can break out into methods? Is there some comment block needed to explain what the method is doing as a whole?

Reply:
I'll break it up, I can move it into descriptive methods, you're right.